demos/combining_metrics.py

```diff
@@ -0,0 +1,116 @@
+# the domain. Metric 2 prescribes the coarse resolution for y<0.3 and y>0.7,
+# but asks for an anisotropic resolution, hc in the x-direction and hf=0.01 in
+# the y-direction, for 0.3<y<0.7
+
+import matplotlib.pyplot as plt
+from firedrake import *
+from animate import *
+mesh = UnitSquareMesh(100, 100)
+P1_ten = TensorFunctionSpace(mesh, "CG", 1)
+metric1 = RiemannianMetric(P1_ten)
+metric2 = RiemannianMetric(P1_ten)
+x, y = SpatialCoordinate(mesh)
+r = 0.2
+hf = 0.01
+hm = 0.025
+hc = 0.1
+anisotropic = as_matrix([[1/hc**2, 0], [0, 1/hf**2]])
+medium = as_matrix([[1/hm**2, 0], [0, 1/hm**2]])
+coarse = as_matrix([[1/hc**2, 0], [0, 1/hc**2]])
+metric1.interpolate(conditional(x < 0.5, medium, coarse))
+metric2.interpolate(conditional(And(abs(x-0.5) < r, abs(y-0.5) < r), anisotropic, coarse))
+metric2.interpolate(conditional(abs(y-0.5) < r, anisotropic, coarse))
+mesh1 = adapt(mesh, metric1)
+mesh2 = adapt(mesh, metric2)
+fig, axes = plt.subplots(figsize=(16, 8), ncols=2)
+triplot(mesh1, axes=axes[0])
+axes[0].set_aspect('equal')
+axes[0].set_title('Mesh based on metric1')
+triplot(mesh2, axes=axes[1])
+axes[1].set_aspect('equal')
+axes[1].set_title('Mesh based on metric2')
+fig.show()
+fig.savefig('combining_two_metrics.jpg')
+
+#
+# .. figure:: combining_two_metrics.jpg
+#    :figwidth: 90%
+#    :align: center
+#
+# We can intersect multiple metric by using the
+# :meth:`~.RiemannianMetric.intersect()` method of a
+# :class:`~.RiemannianMetric`, which intersects the given metric with another,
+# or multiple, provided metrics. We therefore first make a copy of metric1 and
+# then intersect it with metric2:
+
+intersected_metric = metric1.copy(deepcopy=True)
+intersected_metric.intersect(metric2)
+mesh_intersected = adapt(mesh, intersected_metric)
+fig, axes = plt.subplots(figsize=(16, 16))
+triplot(mesh_intersected, axes=axes)
+axes.set_aspect('equal')
+axes.set_title('Mesh based on intersected metric')
+fig.show()
+fig.savefig('combining_intersection.jpg')
+
+#
+# .. figure:: combining_intersection.jpg
+#    :figwidth: 90%
+#    :align: center
+#
+# As we can observe, in every region the resolution respsects the minimum
+# resolution in all directions asked for by both metrics. For example, in the
+# region x<0.5, 0.3<y<0.7 the resolution in the x-direction is hf=0.01, as
+# required by metric2, but in the y-direction the resolution is hm=0.02 as
+# required by metric1.
+#
+# If instead we average the metrics using the
+# :meth:`~.RiemannianMetric.average()` method,
+
+averaged_metric = metric1.copy(deepcopy=True)
+averaged_metric.average(metric2)
+mesh_averaged = adapt(mesh, averaged_metric)
+fig, axes = plt.subplots(figsize=(16, 16))
+triplot(mesh_averaged, axes=axes)
+axes.set_aspect('equal')
+axes.set_title('Mesh based on averaged metric')
+fig.show()
+fig.savefig('combining_averaging.jpg')
+
+#
+# .. figure:: combining_averaging.jpg :figwidth: 90% :align: center
```

Comment:
I think these might need to be on separate lines? The image doesn't appear when I build the html.